src/rust/engine/process_executor/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+use std::io::Error;
+use std::collections::{BTreeSet, HashMap};
+use std::process::Command;
+
+///
+/// A process to be executed.
+///
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ExecuteProcessRequest {
+    argv: Vec<String>,
+    /// Ideally this would be a HashMap, but that isn't hashable, and we need to implement Hash to
+    /// be in the node graph.
+    env: Vec<(String, String)>,
+}
+
+impl ExecuteProcessRequest {
+    ///
+    /// Constructs a new ExecuteProcessRequest.
+    ///
+    /// # Arguments
+    ///
+    /// * `argv` - The arguments to execute. The first argument should be an absolute or relative
+    ///            path to the binary to execute. No shell expansion will take place. No PATH lookup
+    ///            will be performed unless a PATH environment variable is specified.
+    /// * `env` - The environment variables to set for the execution. No other environment variables
+    ///           will be set (except possibly for an empty PATH variable).
+    ///
+    pub fn new(argv: Vec<String>, env: HashMap<String, String>) -> ExecuteProcessRequest {
+        let keys: BTreeSet<String> = env.keys().map(|s| s.to_string()).collect();
```

Comment:
Does this do it?:
```rust
let env_tuples: Vec<(String, String)> = env.iter().collect()
```

With maybe a `map` inline to `clone` the `&String`s.

Reply:
Cute :) The enormity of the magic of .collect() hasn't sunk in to me yet.

Done. Also, added a test to show why sorting matters.